extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/BasicAuthenticationMechanism.java

```diff
@@ -172,8 +172,18 @@ private static void clear(final char[] array) {
                 return CompletableFuture.completedFuture(false);
             }
         }
-        context.response().headers().set(HttpHeaderNames.WWW_AUTHENTICATE, challenge);
-        context.response().setStatusCode(HttpResponseStatus.UNAUTHORIZED.code());
+        context.response().headers().set(challengeHeader(), challengeContent());
```

Comment:
I think all these changes to the mechanisms will need to be changed to support multiple authentication mechanisms. 

Also the API needs to be changed to allow returning an AuthResult with all the info, as this is much more flexible

Reply:
I think, with the current changes, it should be easier to support multiple authentication mechanisms, but I'm not entirely sure how would that work so I'm not certain that's what you meant.
By AuthResult did you mean something like the added `ChallengeData`?